packages/nuxt/src/pages/runtime/plugins/router.ts

```diff
@@ -104,6 +104,9 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
       // so we need to update the route manually
       if (to.matched[0]?.components?.default === from.matched[0]?.components?.default) {
         syncCurrentRoute()
+      } else {
+        // assign toValue to avoid reactivity
+        Object.assign(toValue(_route), router.currentRoute.value) 
```

Comment:
I'm not sure this is the correct fix (but do tell me what you're thinking as I could be wrong here). This is omitted initially as the suspense resolution _should_ update the route (and I think we always want it to be reactive).

Reply:
In that case, should we always sync the route ?

Reply:
We should sync the route, but if we do it here when suspense is involved, it will be too early and update the view before the setup function of the new route has finished

Reply:
I think assigning this way shouldn't update the view since vue won't trigger reactivity. Is there another way of fixing it ?